### ISSUE TITLE
pypi.fallback = cache is raising an exception

### ISSUE THREAD

**Issue by Benoss** [mathcamp/pypicloud#41]
Just installed localy
changed the config to pypi.fallback = cache
```
pypi.default_read = everyone
pypi.default_write = authenticated
pypi.cache_update = everyone
pypi.fallback_url = http://pypi.python.org/simple
pypi.fallback = cache
pypi.storage = file
storage.dir = %(here)s/packages
db.url = sqlite:///%(here)s/db.sqlite
```

Install Django:
```
pip install -i http://localhost:6543/pypi/ Django
Collecting Django
  DEPRECATION: Failed to find 'Django' at http://localhost:6543/pypi/django/. It is suggested to upgrade your index to support normalized names as the name in /simple/{name}.
  Could not find any downloads that satisfy the requirement Django
  No distributions at all found for Django
```

Error log on the server:
```
ERROR 2015-01-22 14:07:50,619 [waitress] Exception when serving /pypi/django/
Traceback (most recent call last):
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/waitress/channel.py", line 337, in service
    task.service()
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/waitress/task.py", line 173, in service
    self.execute()
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/waitress/task.py", line 392, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/router.py", line 242, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/router.py", line 217, in invoke_subrequest
    response = handle_request(request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/tweens.py", line 21, in excview_tween
    response = handler(request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid_tm/__init__.py", line 95, in tm_tween
    reraise(*exc_info)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid_tm/__init__.py", line 76, in tm_tween
    response = handler(request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/router.py", line 163, in handle_request
    response = view_callable(context, request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/config/views.py", line 329, in attr_view
    return view(context, request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/config/views.py", line 305, in predicate_wrapper
    return view(context, request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid/config/views.py", line 355, in rendered_view
    result = view(context, request)
  File "/home/benoit/.virtualenvs/pypicloud/local/lib/python2.7/site-packages/pyramid_duh/view.py", line 181, in slash_redirect
    return fxn(*args)
  File "/home/benoit/git/pypicloud/pypicloud/views/simple.py", line 74, in package_versions
    filename = posixpath.basename(dist.source_url)
AttributeError: 'dict' object has no attribute 'source_url'
```

switching back to pypi.fallback = redirect is working but
```
pip install -i http://localhost:6543/pypi/ Django
Collecting Django
  Using cached Django-1.7.3-py2.py3-none-any.whl
Installing collected packages: Django

Successfully installed Django-1.7.3

```

Another question, is there anyway to have 2 fallback urls? for example pypi and gemfury ?

**Comment by stevearc** [comment]
I suspect this is related to distlib. I'm using it very...optimistically, and if they changed the implementation of the scraper in a recent release then it could cause this. I'll look into it when I get a chance (hopefully this weekend).

In the current implementation, you can only provide a single fallback url.

**Comment by stevearc** [comment]
I seem to be unable to reproduce your crash. Could you tell me what version of python you're using and give me the output of `pip freeze`?

**Comment by stevearc** [comment]
Fixed in 0.2.10 release
